### ISSUE TITLE
How to train model for multiple class

### ISSUE THREAD

**Issue by MaarufB** [ShawnHymel/tflite-speech-recognition#7]
Hi sir this is indeed a brilliant project. I'm just wondering on how can I train the model for multiple classes. Thanks.

**Comment by ShawnHymel** [comment]
You need to change the labels for your dataset to be more than 0 and 1 (e.g. 0, 1, 2, etc.) for your classes. You'd then need to change the model to have that many neurons in the final layer and use categorical cross-entropy for your loss function. Instead of sigmoid, you'd want softmax for the final activation.

That should give you a pretty good start--it's actually quite involved to change it to support multiple classes. Instead, I highly recommend checking out my updated project here: [https://github.com/ShawnHymel/ei-keyword-spotting](https://github.com/ShawnHymel/ei-keyword-spotting). It uses Edge Impulse to do the training, which will give you a model and library that should work across a number of microcontrollers and Linux SBCs.

**Comment by MaarufB** [comment]
Thank you so much. I was able to train the model with multiple classes, however when I used it on Google coral the prediction is not accurate but on windows it is really accurate. Do you have any ideas on how to make it accurate on Coral Dev? This is my repo for the project, I will edit it and use your repo as my reference. Thank you so much. I learn a lot from you.

**Comment by ShawnHymel** [comment]
I honestly don't know much about the Google Coral. My guess is that maybe the Coral used a quantized model whereas it was floating point on Windows, which resulted in some quantization error. I'm not entirely sure, though.
